parser.y

```diff
@@ -2385,7 +2383,43 @@ IndexColNameList:
 		$$ = append($1.([]*ast.IndexColName), $3.(*ast.IndexColName))
 	}
 
+IndexLockAndAlgorithmOpt:
```

Comment:
could you save `lock` infomation in CreateIndexStmt please?

Reply:
L1394:
```
| "ALGORITHM" EqOpt AlterAlgorithm
	{
		// Parse it and ignore it. Just for compatibility.
		$$ = &ast.AlterTableSpec{
			Tp:    		ast.AlterTableAlgorithm,
			Algorithm:	$3.(ast.AlterAlgorithm),
		}
	}
```
could we reuse this code?

Reply:
the name of `ast.AlterAlgorithm`  is no longer suitable, maybe we can call it `AlgorithmType`?